chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -570,6 +608,10 @@ class Bundle(implicit compileOptions: CompileOptions) extends Record {
         Builder.exception(s"Parameterized Bundle ${this.getClass} needs cloneType method")
         this
     }
+
+    require(!newBundle.isLit, s"Bundle ${this.getClass} has a bad cloneType: it should never return a literal")
```

Comment:
This require won't fire if the user overrides cloneType (essentially always given your current examples), right?